gluten-core/src/main/scala/io/glutenproject/execution/BatchScanExecTransformer.scala

```diff
@@ -48,6 +50,8 @@ class BatchScanExecTransformer(
   override def filterExprs(): Seq[Expression] = scan match {
     case fileScan: FileScan =>
       fileScan.dataFilters ++ pushdownFilters
+    case scan if scan.getClass.getSimpleName == "SparkBatchQueryScan" =>
```

Comment:
Comparing simple name may be dangerous. What if Delta Lake (or other formats) has the same class name

Reply:
> Comparing simple name may be dangerous. What if Delta Lake (or other formats) has the same class name

https://github.com/oap-project/gluten/blob/ff79df0084aec978161e99359c3d4dd58376ec45/gluten-core/src/main/scala/io/glutenproject/execution/BatchScanExecTransformer.scala#L127

You are right, it will be more accurate to use a name containing package name, but basically the scan names of different formats are different. Gluten uses simple name when judging the scan type.